pandas/tests/frame/test_arithmetic.py

```diff
@@ -1536,3 +1536,14 @@ def test_dataframe_blockwise_slicelike():
 
         expected = pd.DataFrame({i: left[i] + right[i] for i in left.columns})
         tm.assert_frame_equal(res, expected)
+
+
+def test_dataframe_operation_with_non_numeric_types():
+    # GH #22663
+    df = pd.DataFrame([[1, 2], [4, 5]], columns=list("ab"))
+    result = df + pd.Series([-1], index=list("a"))
+    expected = pd.DataFrame([[0.0, np.nan], [3.0, np.nan]], columns=list("ab"))
+    tm.assert_frame_equal(result, expected)
+    df["b"] = "b"
+    expected = df + pd.Series([-1], index=list("a"))
+    tm.assert_frame_equal(result, expected, check_dtype=False)
```

Comment:
don't use check_dtype

Reply:
I believe it is necessary, because of the column filled with `NaN` values, which is unclear about `dtype`. Here is the test exlained:
```
result = pd.DataFrame([[1, "b"], [4, "b"]], columns=list("ab")) + pd.Series([-1], index=list("a"))
expected = pd.DataFrame([[0.0, np.nan], [3.0, np.nan]], columns=list("ab"))
print(result)
print(result.dtypes)
print(expected)
print(expected.dtypes)
```
outputs are:
```
     a    b
0  0.0  NaN
1  3.0  NaN
a    float64
b     object
dtype: object
     a   b
0  0.0 NaN
1  3.0 NaN
a    float64
b    float64
dtype: object
```

The dtype of column `result['b']` is `object`, whereas `expected['b']` is `float64`. I believe the default `dtype` is `float64` which would raise an exception if I do check dtype.

Reply:
Is it possible to construct `expected` with the correct dtypes instead of using `check_dtype=False`?

Reply:
Yes it is possible, however, I need to add `astype` to both `result` and `expected` as the 2 tests handle with different dtypes (int / float / str + undefined np.nan).
Is it OK this way ?

Reply:
I'm not sure about changing `result` - can't you parametrize the test over the dtype to assign to `b`?

Reply:
OK, I got it. A new PR is going on : to simplify the tests comprehension, I've only worked with `float64` data + I've parametrize a new input data for col `b`